src/main/java/com/aspectsecurity/contrast/contrastjenkins/VulnerabilityTrendStep.java

```diff
@@ -338,19 +349,26 @@ public Void run() throws AbortException, InterruptedException {
                             VulnerabilityTrendHelper.getAgentTypeFromString(step.getAgentType()));
 
                     if(app == null) {
-                        String errorMessage = String.format("Application with [name = %s, agentType = %s] not found.", step.getApplicationName(), step.getAgentType());
-                        if(teamServerProfile.isFailOnWrongApplicationId()) {
-                            throw new AbortException(errorMessage);
-                        } else {
-                            VulnerabilityTrendHelper.logMessage(taskListener, errorMessage);
+                        String errorMessage = String.format(CONTRAST_ERROR_PREFIX + "Application with [name = %s, agentType = %s] not found.", step.getApplicationName(), step.getAgentType());
+                        VulnerabilityTrendHelper.logMessage(taskListener, errorMessage);
+                        if(teamServerProfile.isApplyVulnerableBuildResultOnContrastError()) {
+                            Result profileVulnerableBuildResult = Result.fromString(teamServerProfile.getVulnerableBuildResult());
+                            build.setResult(profileVulnerableBuildResult);
                         }
+                        return null;
                     } else {
                         step.setApplicationId(app.getId());
                         VulnerabilityTrendHelper.logMessage(taskListener, "Fetched Application : [name = '"+step.getApplicationName()+"', displayName = '"+app.getName()+"', agentType='"+app.getLanguage()+"'] with ID: ["+step.getApplicationId()+"]");
                     }
                 } catch (UnauthorizedException | IOException e) {
+                    String errorMessage = CONTRAST_ERROR_PREFIX + "Unable to retrieve information from TeamServer.";
+                    VulnerabilityTrendHelper.logMessage(taskListener, errorMessage);
                     VulnerabilityTrendHelper.logMessage(taskListener, e.getMessage());
-                    throw new AbortException("Unable to retrieve information from TeamServer.");
+                    if(teamServerProfile.isApplyVulnerableBuildResultOnContrastError()) {
```

Comment:
Can we throw this if statement into a method since it gets reused quite a bit? The same for in VulnerabilityTrendRecorder, but with that if statement.

Reply:
updated